crates/ruff_server/src/server/api/requests/format.rs

```diff
@@ -23,25 +27,73 @@ impl super::BackgroundDocumentRequestHandler for Format {
     }
 }
 
+/// Formats either a full text document or each individual cell in a single notebook document.
+pub(super) fn format_full_document(snapshot: &DocumentSnapshot) -> Result<Fixes> {
+    let mut fixes = Fixes::default();
+
+    if let Some(notebook) = snapshot.query().as_notebook() {
```

Comment:
Nit: Would it make sense to store the result of  `snapshot.query()` in a variable. It's used in multiple places and feels fairly repetitive.